### ISSUE TITLE
UserWarning: Private repos not supported. Upon checking SCT install

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#4712]
After installing SCT 6.5, the last 'checking' step shows me this:

````
Check if dipy is installed........................../Users/julien/code/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_check_dependencies.py:164: UserWarning: Private repos not supported. Skipping.
  for req in requirements.parse(requirements_txt):
````

Not a big deal, but we should remove is not too complicated

**Comment by joshuacwnewton** [comment]
The cause of this warning is very simple. In the following function call:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b083497faacb5ab74209c28b30a92cc85fc197b6/spinalcordtoolbox/scripts/sct_check_dependencies.py#L164

The function `requirements.parse` contains the following logic:

```python
        elif line.startswith('-f') or line.startswith('--find-links') or \
                line.startswith('-i') or line.startswith('--index-url') or \
                line.startswith('--extra-index-url') or \
                line.startswith('--no-index'):
            warnings.warn('Private repos not supported. Skipping.', stacklevel=2)
            continue
```

Which gets triggered for the following line in our requirements.txt:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/89b1ffcbc649cb0f90573cb37aa378f1a3cd3b08/requirements.txt#L12

In other words, it's a spurious warning that can be explicitly ignored. This is easily fixed, and I'll open up a quick PR.
